vignettes/messydates.Rmd

```diff
@@ -0,0 +1,151 @@
+
+# Working with messydates: The size of the popuklation of the Roman Empire
+
+Several historical facts are contested. Take, for example, the population sizes in the Roman Empire, which has been a topic of heated debates among some academics. Various accounts diverge in how they estimate population size and, therefore, the conclusions about the Roman population at different points in time. Below, we exemplify how \pkg{messydates} facilitate the annotation and the conversion of historical dates for analysis on the dataset of the high and low estimations for the population of mainland Italy during the Roman Empire included in the package.
+
+```{r data, warning=FALSE}
+library(messydates)
+roman_population # For more information on the dataset please run `?roman_population`.
+```
+
+## Coerce to messydates
+
+The function `as_messydate()` handles the annotation and coercion to 'mdate' class.
+We can, for instance, coerce the date variable in the Roman population data to 'mdate'.
+
+```{r messy, warning=FALSE}
+roman_population$period <- as_messydate(roman_population$period)
+roman_population$period
+```
+
+## Annotate
+
+Some datasets have, for example, an arbitrary cut off point for start and end points, 
+but these are often coded as precise dates when they are not necessarily the real start or end dates.
+The annotate functions helps annotate uncertainty and approximation to dates.
+Inaccurate start or end dates can be represented by an ".." affix
+indicating "on or before", if used as a prefix,
+or indicating "on or after", if used as a suffix.
+Approximate dates are indicated by adding a `~` to year,
+month, or day components, as well as groups of components or whole dates
+to estimate values that are possibly correct.
+Day, month, or year, uncertainty can be indicated by adding a `?`
+to a possibly dubious date or date component.
+
+For example, we can add these annotations to the 'mdate' variable in the Roman population data.
+
+```{r annotate, warning=FALSE}
+ifelse(roman_population$period == "-0200..-0100", on_or_before(roman_population$period), roman_population$period)
+ifelse(roman_population$period == "0150..0200", on_or_after(roman_population$period), roman_population$period)
+ifelse(roman_population$period == "-0100..0001", as_approximate(roman_population$period), roman_population$period)
+ifelse(roman_population$period == "0001..0150", as_uncertain(roman_population$period), roman_population$period)
+```
+
+## Expand
+
+Expand functions transform date ranges, sets of dates, and unspecified or 
+approximate dates (annotated with '..', '{ , }', 'XX' or '~') into lists of dates.
+As these dates may refer to several possible dates,
+the function "opens" these values to include all the possible dates implied.
+
+The ranges in the 'mdate' variable in the Roman population data can be also expanded.
```

Comment:
We need to add more information about the default window size here.

Reply:
I have changed the default 'approx_range' argument for `expand()` approximate dates or uncertain dates from 3 to 0. This means these are not expanded unless users explicitly want to do so. The reason why I believe this is a better choice is because, first, we had an arbitrary number for the expansion and, second, users might not understand why certain dates became ranges and why some returned values are so different than the value entered. In any case, I also added a more informative message in the function for users to refer to the "approx_range" argument if they want to expand approximate dates.